Godeps/_workspace/src/k8s.io/kubernetes/pkg/proxy/hybrid/proxier.go

```diff
@@ -0,0 +1,130 @@
+package hybrid
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/proxy/config"
+	"k8s.io/kubernetes/pkg/proxy"
+	"k8s.io/kubernetes/pkg/proxy/userspace"
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/util/unidling"
+
+	"github.com/golang/glog"
+)
+
+type Proxier struct {
+	unidlingProxy        *userspace.Proxier
+	unidlingLoadBalancer config.EndpointsConfigHandler
+	mainProxy            proxy.ProxyProvider
+	mainLoadBalancer     config.EndpointsConfigHandler
+
+	serviceConfig *config.ServiceConfig
+
+	// TODO(directxman12): figure out a good way to avoid duplicating this information
+	// (it's saved in the individual proxies as well)
+	usingUserspace map[types.NamespacedName]bool
```

Comment:
style nit: `map[types.NamespacedName]struct{}`. You use the `_, found := x[key]` lookup style anyway.